### ISSUE TITLE
[Bug] [UWP] Memory Leak When Changing the ItemsSource on a CollectionView

### ISSUE THREAD

**Issue by tstrickler4** [xamarin/Xamarin.Forms#13928]
### Description

Changing the ItemsSource on a CollectionView in UWP doesn't free up resources, thus causing a memory leak. I've confirmed that the issue occurs in both Debug and Release modes on UWP. I haven't tested Android or iOS, but I've included Android and iOS projects for convenience.

### Steps to Reproduce

1. Build the included solution.
2. Use the two buttons at the bottom to switch between two different Lists for the ItemsSource.
3. Observe the memory usage continues to grow each time it's switched.

### Expected Behavior

Free up memory after the ItemsSource is switched.

### Actual Behavior

Memory usage continues to grow each time the ItemsSource is switched.

### Basic Information

- Version with issue: Xamarin.Forms v5.0.0.2012
- Last known good version: Unknown
- Platform Target Frameworks: 
  - UWP:  Windows 10, version 2004 (10.0; Build 19041)

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
Microsoft Visual Studio Community 2019
Version 16.8.4
VisualStudio.16.Release/16.8.4+30907.101
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA890
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.8.557.25636
ASP.NET and Web Tools 2019

ASP.NET Core Razor Language Services   16.1.0.2052803+84e121f1403378489b842e1797df2f3f5a49ac3c
Provides languages services for ASP.NET Core Razor.

ASP.NET Web Frameworks and Tools 2019   16.8.557.25636
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   16.8.557.25636
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   16.8.557.25636
Azure Functions and Web Jobs Tools

C# Tools   3.8.0-5.20604.10+9ed4b774d20940880de8df1ca8b07508aa01c8cd
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

JetBrains ReSharper 2020.3.2   Build 203.0.20201229.111204
JetBrains ReSharper package for Microsoft Visual Studio. For more information about ReSharper, visit http://www.jetbrains.com/resharper. Copyright © 2021 JetBrains, Inc.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2019 - v2.9.30924.1

Microsoft Continuous Delivery Tools for Visual Studio   0.4
Simplifying the configuration of Azure DevOps pipelines from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.113+g422d40002e.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.1
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   16.8.43 (00471f8)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   5.8.1
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   16.0.62012.31170
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.21016.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.8.0-5.20604.10+9ed4b774d20940880de8df1ca8b07508aa01c8cd
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   16.8.0-beta.20507.4+da6be68280c89131cdba2045525b80890401defd
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.8.000.261 (d16-8@bb99248)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   16.8.0.507 (remotes/origin/d16-8@e87b24884)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.8.112 (86385a3)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   11.1.0.26 (d16-8/a36ce73)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: 5e9cb6d
    Java.Interop: xamarin/java.interop/d16-8@79d9533
    ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
    SQLite: xamarin/sqlite/3.32.1@1a3276b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc


Xamarin.iOS and Xamarin.Mac SDK   14.8.0.3 (c51fabee8)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

</details>

### Screenshots

![image](https://user-images.githubusercontent.com/24135691/109827809-adf10f00-7c01-11eb-8eaa-5ea54a3766e9.png)

### Reproduction Link

[MemoryLeakDemo.zip](https://github.com/xamarin/Xamarin.Forms/files/6077009/MemoryLeakDemo.zip)

**Comment by tstrickler4** [comment]
I forgot to mention that this also appears to be a problem with ListViews.

**Comment by PureWeen** [comment]
Closing for now as a duplicate of https://github.com/xamarin/Xamarin.Forms/issues/8928

**Comment by tstrickler4** [comment]
@PureWeen Are you sure this is a duplicate? That one mentions a memory leak as a result of navigating away from a page containing a CollectionView and the event not being unhooked, whereas the problem here is changing the ItemsSource while on the same page. No navigation necessary. If they share the same root cause, then it's fine. I just want to make sure it's not being closed prematurely.

**Comment by ssombat** [comment]
I agreed with @tstrickler4, this is different from #8928. 

@tstrickler4 were you able to find a work around for this?

**Comment by TylerTKiMobility** [comment]
Absolutely not a duplicate. (Not a great solution but it does work) Utilize an ObservableCollection for binding the itemSource property and just do a .Clear() and .Add() to the binded property instead of setting it to a new List/ObservableCollection.

**Comment by YZahringer** [comment]
Fixed by #14780
